cmd/oceantv/broadcast_hardware_machine.go

```diff
@@ -61,6 +109,36 @@ func (s *hardwareStarting) timedOut(t time.Time) bool {
 
 func (s *hardwareStarting) reset(time.Duration) {}
 
+type hardwareRecoveringVoltage struct {
+	stateFields
+	stateWithTimeoutFields
+}
+
+func newHardwareRecoveringVoltage(ctx *broadcastContext) *hardwareRecoveringVoltage {
+	s := newStateWithTimeoutFields(ctx)
+	s.Timeout = time.Duration(sanatisedVoltageRecoveryTimeout(ctx)) * time.Hour
+	return &hardwareRecoveringVoltage{
+		stateWithTimeoutFields: s,
+	}
+}
+
+func (s *hardwareRecoveringVoltage) enter() {
+	s.LastEntered = time.Now()
+}
+
+func sanatisedVoltageRecoveryTimeout(ctx *broadcastContext) int {
+	// If VoltageRecoveryTimeout is not set, default to 4 hours.
+	if ctx.cfg.VoltageRecoveryTimeout == 0 {
```

Comment:
Where is VoltageRecoveryTimeout set?

Reply:
This will be available under the advanced broadcast settings.